### DISCUSSION TITLE
Per-package time based cooldown

### DISCUSSION THREAD

**Discussion by ntkoopman** [renovatebot/renovate#18298] [Suggest an Idea]
Would it be possible to add an option to not create PRs for dependencies that have been updated recently already?

For example, to limit updates to once every 2 weeks, except for major updates:
```
{
  "packageRules": [
    {
      "matchUpdateTypes": ["minor", "patch"],
      "cooldown": "P14D",
    }
  ]
}
```

If necessary, the last update time could be stored in the repository somewhere.

(this would be different from `stabilityDays` which delays PR creating instead of rate limiting them)

**Comment by rarkins** [comment]
There's a few edge case considerations to think of first. For example:
- Once a PR is created, should it be never updated? This is hard to do if for example it becomes conflicted or behind base branch - it would need to stay conflicted
- What to do if you group together and dependencies have competing cooldown periods?

**Reply by ntkoopman** [reply]
For the first one, as long as the PR is not merged I wouldn't expect it to count for the cooldown period, so it would keep updating indefinitely until you merge it. If you have separate PRs for major/minor versions, I would expect the other PR to be closed if you merge one (or if the cooldown is extended for some other reason, like updating the config). If you don't merge the PR, it wouldn't reset the cooldown period so it would work exactly the same as it does now.

For the second one, I think it probably doesn't matter too much as long as it's documented somewhere. I think you could make a case for taking both the minimal cooldown period and the maximum one. I would assume most people would configure the cooldown period on the same rule as the group if they choose to use this option.

**Reply by rarkins** [reply]
So the cooldown period only affects *creation* of PRs and is essentially ignored once a PR is open?

**Reply by ntkoopman** [reply]
That is what it would look like, but probably not how it would be implemented? Since a PR would only be created after the cooldown period, any updates to the PR would by definition also be after the cooldown period. The cooldown period would only reset when you apply an update.

I just noticed it might be unclear that when I mentioned "last updated", I meant last updated in the local project, not last updated upstream.
